### DISCUSSION TITLE
[Question] Question about deformable body material

### DISCUSSION THREAD

**Discussion by yuezhezhang** [isaac-sim/IsaacLab#1136] [Q&A]
### Question

Hi,

I try to test the behaviors of deformable objects in file [run_deformable_object](https://github.com/isaac-sim/IsaacLab/blob/main/source/standalone/tutorials/01_assets/run_deformable_object.py).

As I searched [online](https://www.physicsforums.com/threads/steel-youngs-modulus-poisson-ratio.523754/) the poisson ratio and Young's module of steel should be 0.3 and 200GPa, but when I set `poissons_ratio=0.3, youngs_modulus=2e11`, it does not seem like steel:

https://github.com/user-attachments/assets/9163feef-367e-4905-b6ec-5562b7105f1c

I also tried to pick the object, and the gripper also seemed to smash it. 

Do you know how may I tune the behavior better? Thanks in advance.

**Comment by Mayankm96** [comment]
From what the PhysX team mentioned -- For higher stiffness, you can:

- lower the voxel mesh resolution (=faster convergence)
- lower the time step
- increase the iteration count
- maybe enable option "external forces per substep" but it does not help that much
- Use the TGS solver (but that's the default solver in kit anyway)

Also a soft body will always be a bit soft, no matter how hard you optimize the solver. For real rigid bodies, just use the PhysX SDK's rigid bodies.

**Reply by yuezhezhang** [reply]
Hi Mayank, 

thanks for the feedback!

I searched the code and found relevant properties of soft body may lie in the functions `modify_deformable_body_properties` of `source/extensions/omni.isaac.lab/omni/isaac/lab/sim/schemas/schemas.py` and `add_physx_deformable_body` of `_isaac_sim/extsPhysics/omni.physx/omni/physx/scripts/deformableUtils.py`. 

Specifically, in `add_physx_deformable_body`, I guess these arguments correspond to what you mentioned:
- `simulation_hexahedral_resolution`: voxel mesh resolution
- `solver_position_iteration_count`: iteration count

I guess I am correct since IsaacSim 4.2.0 lowered `simulation_hexahedral_resolution` from 10 to 1 and updated `solver_position_iteration_count` from None to 10. This change is in line with your suggestion. 

However, sadly, it is still a bit soft though I tried to tune the `poissons_ratio, youngs_modulus, dynamic_friction` as [this](https://github.com/isaac-sim/IsaacLab/pull/972).

I may leave this discussion as it is now, if someone has new findings or discoveries, please hit me up!
